mutiny-core/src/storage.rs

```diff
@@ -0,0 +1,436 @@
+    }
+}
+
+impl MutinyStorage for MemoryStorage {
+    fn password(&self) -> Option<&str> {
+        self.password.as_deref()
+    }
+
+    fn set<T>(&self, key: impl AsRef<str>, value: T) -> Result<(), MutinyError>
+    where
+        T: Serialize,
+    {
+        let key = key.as_ref().to_string();
+        let data = serde_json::to_value(value).map_err(|e| MutinyError::PersistenceFailed {
+            source: MutinyStorageError::SerdeError { source: e },
+        })?;
+        let mut map = self
+            .memory
+            .try_write()
+            .map_err(|e| MutinyError::write_err(e.into()))?;
+        map.insert(key, data);
+
+        Ok(())
+    }
+
+    fn get<T>(&self, key: impl AsRef<str>) -> Result<Option<T>, MutinyError>
+    where
+        T: for<'de> Deserialize<'de>,
+    {
+        let map = self
+            .memory
+            .try_read()
+            .map_err(|e| MutinyError::read_err(e.into()))?;
+
+        match map.get(key.as_ref()) {
+            None => Ok(None),
+            Some(value) => {
+                let data: T = serde_json::from_value(value.to_owned())?;
+                Ok(Some(data))
+            }
+        }
+    }
+
+    fn delete(&self, key: impl AsRef<str>) -> Result<(), MutinyError> {
+        let key = key.as_ref().to_string();
+
+        let mut map = self
+            .memory
+            .try_write()
+            .map_err(|e| MutinyError::write_err(e.into()))?;
+        map.remove(&key);
+
+        Ok(())
+    }
+
+    async fn start(&mut self) -> Result<(), MutinyError> {
+        Ok(())
+    }
+
+    fn stop(&self) {}
+
+    fn connected(&self) -> Result<bool, MutinyError> {
+        Ok(false)
+    }
+
+    fn scan<T>(&self, prefix: &str, suffix: Option<&str>) -> Result<HashMap<String, T>, MutinyError>
+    where
+        T: for<'de> Deserialize<'de>,
+    {
+        let map = self
+            .memory
+            .try_read()
+            .map_err(|e| MutinyError::read_err(e.into()))?;
+
+        Ok(map
+            .keys()
+            .filter(|key| {
+                key.starts_with(prefix) && (suffix.is_none() || key.ends_with(suffix.unwrap()))
+            })
+            .filter_map(|key| {
+                self.get_data(key)
+                    .ok()
+                    .flatten()
+                    .map(|value: T| (key.to_owned(), value))
+            })
+            .collect())
+    }
+
+    fn insert_mnemonic(&self, mnemonic: Mnemonic) -> Result<Mnemonic, MutinyError> {
+        self.set_data(MNEMONIC_KEY, mnemonic.clone())?;
+
+        Ok(mnemonic)
+    }
+
+    fn get_mnemonic(&self) -> Result<Mnemonic, MutinyError> {
+        self.get_data(MNEMONIC_KEY)?.ok_or(MutinyError::NotFound)
+    }
+
+    async fn import(_json: Value) -> Result<(), MutinyError> {
+        Ok(())
+    }
+
+    async fn clear() -> Result<(), MutinyError> {
+        Ok(())
+    }
+}
+
+// Dummy implementation for testing or if people want to ignore persistence
+impl MutinyStorage for () {
```

Comment:
Wait, what does this mean? lol. 

If it's just for tests, can we hide behind a test flag? And for people that want to ignore persistences, can we still just keep it with an in memory store? We'll have to do this for incognito browsers or for brave in protection mode.

Reply:
Yeah this is basically for tests or something. Just a dummy implementation so if someone wants to get started without having to implement storage first.

Reply:
Gotcha, was unfamily with the `for ()` part. I still think it's good to do one that's in memory only but still actually usable but that can come later.

Reply:
There is a `MemoryStorage` I implemented as well